sysml.library/Domain Libraries/Geometry/ShapeItems.sysml

```diff
@@ -0,0 +1,249 @@
+package ShapeItems {
+	private import ScalarValues::String;
+	private import ScalarValues::Boolean;
+	private import UnitsAndScales::CoordinateTransformation;
+	private import ISQ::*;
+	private import SI::*;
+	private import Collections::KeyValuePair;
+	private import Collections::Map;
+	private import Objects::*;
+	private import Items::*;
+	private import SequenceFunctions::union;
+	private import CollectionFunctions::contains;
+	private import SequenceFunctions::isEmpty;
+	connection def WithinBoth :> Occurrences::WithinBoth { }
+	
+	/**
+	 * Most general shape.	Solids are objects with these as space boundary.	
+	 */
+	abstract item def Shell :> Item, StructuredSpaceObject, Surface	 {
+	}
+
+	/**
+	* Straight line.
+	*/
+	item def Line :> Curve {
+		attribute :>> length;
+
+		attribute :>> outerSpaceDimension = 1;
+	}
+
+	/**
+	 * Flat surface.
+	 */
+	item def PlanarSurface :> Surface {
+		attribute :>> area;
+		attribute :>> outerSpaceDimension = 2;
+	}
+
+	item def Circle :> Shell {
+		attribute radius :>> radius;
+
+		item :>> faces [0];
+		item :>> edges [1] {
+			attribute circumference :>> length = Circle::radius * 3.1415926535898;
```

Comment:
Should use TrigFunctions::pi.

Reply:
Thx, I fixed this in my local copy.  Figured I'd wait for HP/your reply on the next comment to push.

Reply:
Done.